demos/bookstore/app/middleware/cart.ts

```diff
@@ -0,0 +1,10 @@
+import type { Middleware } from '@remix-run/fetch-router'
+import { createCartIfNotExists } from '../models/cart.ts'
+
+/**
+ * Middleware that ensures the user session has an associated cart
+ * To be used on any route that mutates the cart
+ */
+export const ensureCart: Middleware = async ({ session }) => {
+  createCartIfNotExists(session)
```

Comment:
Middleware used by the cart API routes to ensure we always have a cart available if we're trying to mutate the cart